### PR TITLE
fix #3078: adding more javadocs around operations

### LINKED ISSUE #3078: Item passed in is modified by replace or patch calls
The item passed in can have the resourceVersion set as a side effect of replace or patch calls.  Is it expected and/or should it be part of the javadoc?

### PR DESCRIPTION
**Pull request by shawkins**
## Description
Based upon the discussion on #3078 more javadocs seemed in order to help clarify usage and the possibility of items being modified.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Editable.java [hunk 1]
```diff
@@ -22,11 +22,37 @@
...
+     * 
+     * @param visitors to modify the Builder
+     * @return returns deserialized version of api server response
+     */
     T edit(Visitor... visitors);
 
+    /**
+     * Issues a JSON patch against the item based upon the changes made to the object by the visitor.
+     * 
+     * @param visitor to modify the Builder
```
Comment (rohanKanojia) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Editable.java:46]: `@param` entry for visitorType is missing

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Editable.java [hunk 2]
```diff
@@ -22,11 +22,37 @@
...
+     * @return returns deserialized version of api server response
+     */
     <V> T edit(Class<V> visitorType, Visitor<V> visitor);
 
+    /**
+     * Issues a JSON patch against the item based upon the changes made by the
```
Comment (rohanKanojia) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Editable.java:52]:
```suggestion
     * Issues a JSON patch against the item based upon the changes made by the function provided as argument
```

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Replaceable.java [hunk 3]
```diff
@@ -17,6 +17,19 @@
 
 public interface Replaceable<T> {
 
+  /**
+   * Replace the server's state with the given item. The resourceVersion and other fields of the item may be modified by this call.
+   * 
+   * <p>If {@link Lockable#lockResourceVersion(String)} has been used to lock the resourceVersion, 
+   * this operation is effectively a single update attempt against that version.
+   * <p>If {@link Lockable#lockResourceVersion(String)} has not been called, this operation will attempt
```
Comment (rohanKanojia) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Replaceable.java:25]:
```suggestion
   * <p>If {@link Lockable#lockResourceVersion(String)} has not been called, this operation
```

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/HasMetadataOperation.java [hunk 4]
```diff
@@ -140,6 +140,9 @@ protected T replace(T item, boolean status) {
     String fixedResourceVersion = getResourceVersion();
     Exception caught = null;
     int maxTries = 10;
+    if (item.getMetadata() == null) {
+      item.setMetadata(new ObjectMeta());
+    }
```
Comment (rohanKanojia) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/HasMetadataOperation.java:145]: Sorry if I'm missing context. Why is this change required?

Comment (shawkins) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/HasMetadataOperation.java:145]: Just to avoid the later npe - but it could be handled via an arg check or left out if you want.

**Review by manusa — APPROVED** [review]
LGTM, thx!
